### PR TITLE
Add `names` field to genes, metabolites and reaction types.

### LINKED ISSUE #565: Add `name` field to `StandardModel`
Currently `StandardModel` does not import the `name` of a reaction. This could be useful during reconstruction... Fix.

### PR DESCRIPTION
**Pull request by stelmo**
Also added accessors for JSONModel and StandardModel (SBMLModel is the dark horse...)

Closes #565 when complete

### CODE REVIEW

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit 18616f7b requires formatting!

Required formatting changes summary:
```
 src/base/types/JSONModel.jl | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)
```

File: src/base/types/Gene.jl [hunk 1]
```diff
@@ -4,15 +4,17 @@ Gene struct.
...
 mutable struct Gene
     id::String
+    name::Maybe{String}
     notes::Notes
     annotations::Annotations
 
-    Gene(id::String = ""; notes = Notes(), annotations = Annotations()) =
-        new(id, notes, annotations)
+    Gene(id::String = ""; name = nothing, notes = Notes(), annotations = Annotations()) =
+        new(id, name, notes, annotations)
```
Comment (exaexa) [inline src/base/types/Gene.jl:19]: might be interesting to incorporate what we started using in SBML, basically there's a macro `@kwdef` that does this correctly:
https://github.com/LCSB-BioCore/SBML.jl/blob/master/src/structs.jl#L109

Not necessary now, just dropping the idea here. :D

File: src/base/types/Reaction.jl [hunk 2]
```diff
@@ -1,6 +1,7 @@
 """
     mutable struct Reaction
         id::String
+        name::Maybe{String}
```
Comment (exaexa) [inline src/base/types/Reaction.jl:4]: note to self: names should also be supported by SBML models (SBML.jl supports names now)

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit ef05d600 is formatted properly.

File: src/base/types/Reaction.jl (hunk 2 above)
Comment (exaexa) [reply src/base/types/Reaction.jl:4]: in #568

**Review by exaexa — APPROVED** [review]

**Pull request merged**